bootstrap_sdk

```diff
@@ -99,41 +109,63 @@ cp "${BUILD_LIBRARY_DIR}/toolchain_util.sh" "${ROOT_OVERLAY}/tmp"
 # Stage 1 uses "known-good" ebuild repos (both coreos-overlay and portage-stable)
 #  to build a minimal toolchain (USE="-*") for stage 2.
 #
-# No package updates must happen in stage 1, so we use
-# - the portage-stable repo included with the current SDK (from the SDK chroot's
-#    /var/lib/gentoo/repos/gentoo), and
-# - the coreos-overlay repo version listed in the current SDK's manifest.
-#    Unfortunately coreos-overlay is not included in the SDK chroot at all; this
-#    will be addressed in the future.
-#
-# "Current SDK" refers to the SDK we entered with 'cork enter', i.e. the SDK
-#   we run ./bootstrap_sdk in.
+# No package updates must happen in stage 1, so we use the portage-stable and
+#  coreos-overlay repos included with the current SDK (from the SDK chroot's
+#  /var/lib/gentoo/repos/). "Current SDK" refers to the SDK we entered with
+#  'cork enter', i.e. the SDK we run ./bootstrap_sdk in.
 #
 # Using ebuilds from the above mentioned sources will ensure that stage 1 builds
 #  a minimal stage 2 from known-good ebuild versions - the same ebuild versions
 #  that were used to build the very SDK we run ./bootstrap_sdk in.
 #
-# NOTE that stage 1 lacks proper isolation and will link all packages built for stage 2
-#  against its own seed libraries ("/" in the catalyst chroot) instead of against libraries
+# This is needed because stage 1 lacks proper isolation and will link all packages built for
+#  stage 2 against its own seed libraries ("/" in the catalyst chroot) instead of against libraries
 #  installed into the FS root of the stage 2 seed ("/tmp/stage1root" in the catalyst chroot).
 #  This is why we must prevent any updated package ebuilds to "leak" into stage 1, hence we use
 #  "known good" ebuild repo versions outlined above.
+#
+# In special circumstances it may be required to circumvent this and use git
+#  refs of either (or both) portage and overlay. The command line options
+#  --stage1-portage-ref and --stage1-overlay-ref may be used to specify
+#  a git ref known to work for stage1. In that case the stage1 seed (i.e. the seed SDK)
+#  will be updated prior to starting to build stage 2.
+
+stage_repo() {
+    local repo="$1"
+    local gitref="$2"
+    local dest="$3"
+    local gitname="$repo"
+
+    [ "$gitname" = "gentoo" ] && gitname="portage-stable"
```

Comment:
an `if` statement clearer. In case someone adds `set -e` and `gitname` is not `gentoo`, the execution will stop at this command because the return code is not 0

Reply:
Fixed, good catch!